gnd/src/main/java/com/google/android/gnd/workers/FileDownloadWorkManager.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.workers;
+
+import androidx.lifecycle.LiveData;
+import androidx.work.Constraints;
+import androidx.work.ExistingWorkPolicy;
+import androidx.work.NetworkType;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkInfo;
+import androidx.work.WorkManager;
+
+/** Enqueues file download work to be done in the background. */
+public class FileDownloadWorkManager {
+  private final WorkManager workManager;
+
+  public FileDownloadWorkManager() {
+    this.workManager = WorkManager.getInstance();
+  }
+
+  /**
+   * Enqueues a worker that downloads files when a network connection is available, returning live
+   * data that contains {@link WorkInfo}. Observe the returned live data to do something with the
+   * worker's result.
+   */
+  public LiveData<WorkInfo> enqueueFileDownloadWorker(String url, String filename) {
```

Comment:
Why do we need the WorkInfo?

Reply:
At the moment, I'm passing the FILENAME used by the worker back to callers in case the filename changes during the worker's processing--that way callers don't have to assume the filename they passed is where the file ultimately lives--then again, we should probably just overwrite files with the same name for our use case, so I'll modify this to send an empty result signal instead.

Reply:
I see.. would it make more sense for this worker to do the full flow of downloading the file and writing the current download state/progress to the local db? In that case the UI could be reactive on the db as with other features, rather than needing to interact via returned LiveData.

Reply:
Actually, I think that's a good idea. While it will make the worker itself slightly more complex, I think that will save us huge potential headaches in the rest of the app--much simpler to call a worker to download something and have it take care of everything rather than call a worker to download something and then have to recall to follow up with a db write yourself. Let's try that approach!

Reply:
Actually, I'm not sure we can do this. 

My plan is to implement a db entity for tiles--but to also use this worker to download the geoJSON file, when it's not already on the device. For the former case, there's no tile to write for the DB.

Unless we're planning on just bundling the geoJSON?

Reply:
Well, let's forge ahead assuming we can make the worker handle every aspect of the process--since the GeoJSON is a special case, we'll figure out what to do there separately.

Reply:
+1. For now let's bundle GeoJSON, treating it as configuration for how to access our offline tile source.